projects/commudle-admin/src/app/feature-modules/events/components/user-event-registrations/entry-pass-scan/entry-pass-scan.component.html

```diff
@@ -0,0 +1,74 @@
+                <source
+                  src="https://json.commudle.com/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBb1FkIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--9255d1e05be35d9fb971bfb38d1a67d8c39c8b0f/success_sound.mp3"
+                  type="audio/mp3">
+              </audio>
+              <audio #incorrectSound>
+                <source
+                  src="https://json.commudle.com/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBb1VkIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--52910f934db167843387e03f3500aeff6be3a366/error_sound.mp3"
+                  type="audio/mp3">
+              </audio>
+            </div>
+            <div class="cameras-list">
+              <nb-form-field>
+                <nb-icon nbPrefix icon="alert-circle-outline"></nb-icon>
+                <nb-select placeholder="Select Camera" [(selected)]="currentDevice"
+                  (selectedChange)="onDeviceSelectChange($event)" status="info" [compareWith]="compareByDeviceId">
+                  <nb-option value="">No Device Selected</nb-option>
+                  <nb-option *ngFor="let device of availableDevices" [value]="device.deviceId">{{ device.label }}
+                  </nb-option>
+                </nb-select>
+              </nb-form-field>
+            </div>
+            <div *ngIf="eventPass" class="user-details-container">
+              <app-user-profile-card-large *ngIf="eventPass.user" [user]="eventPass.user"></app-user-profile-card-large>
+              <p *ngIf="eventPass.user" class="contact-details">
+                <small>{{ eventPass.user.email }}</small>
+              </p>
+              <div *ngIf="eventPass.user" class="social-links">
+                <a *ngIf="eventPass.user.github" href="{{ eventPass.user.github }}">
+                  <fa-icon [icon]="faGithub"></fa-icon>
+                </a>
+                <a *ngIf="eventPass.user.twitter" href="{{ eventPass.user.twitter }}">
+                  <fa-icon [icon]="faTwitter"></fa-icon>
+                </a>
+                <a *ngIf="eventPass.user.linkedin" href="{{ eventPass.user.linkedin }}">
+                  <fa-icon [icon]="faLinkedin"></fa-icon>
+                </a>
+                <a *ngIf="eventPass.user.personal_website" href="{{ eventPass.user.personal_website }}">
+                  <fa-icon [icon]="faInfo"></fa-icon>
+                </a>
+              </div>
+              <p *ngIf="eventPass.unique_code" class="entry-pass-code">Entry Pass: {{ eventPass.unique_code }}</p>
+              <button *ngIf="eventPass" (click)="toggleAttendance()" nbButton outline fullWidth status="success">
+                Mark Attendance
+              </button>
+            </div>
+          </div>
+
+          <ng-container *ngIf="hasDevices === undefined">
+            <h2 class="com-text-center">No cameras available!</h2>
+          </ng-container>
+
+          <ng-conatiner *ngIf="hasPermission === undefined">
```

Comment:
Let's not check against undefined, instead, let's do `!hasPermission`

Reply:
There are two things:
1. Permission pop up came and user haven't clicked on anything i.e. has permission is undefined
2. Permission pop up came and user clicked on block i.e. has permission is false

Reply:
done!